test/bio_base64_test.c

```diff
@@ -6,107 +6,263 @@
  * in the file LICENSE in the source distribution or at
  * https://www.openssl.org/source/license.html
  */
+#include <limits.h>
 #include <stdio.h>
 #include <string.h>
+#include <time.h>
+#include <openssl/bio.h>
 #include <openssl/evp.h>
+#include <openssl/rand.h>
+
+#include "testutil.h"
+
+/* 2048 bytes of "#ooooooooo...\n" + NUL terminator */
+static char gunk[2049];
 
 typedef struct {
+    char *prefix;
     char *encoded;
-    int encoded_len;
-    char *expected_decode;
-    int expected_decode_len;
+    unsigned bytes;
+    char *suffix;
```

Comment:
nit: if you reverse bytes and suffix, you'll eliminate the alignment gap this would otherwise create on 64 bit systems.

Reply:
Sure, but in the test code I prefer the logical sequence of prefix, data, suffix, and the alignment gap is inconsequential.